Enhance Fitbit integration documentation with troubleshooting tips (#43394)
home-assistant/home-assistant.github.io: source/_integrations/fitbit.markdown

Please be aware that Fitbit has very low rate limits, 150 per user per hour. The clock resets at the _top_ of the hour (meaning it is not a rolling 60 minutes). There is no way around the limits. Due to the rate limits, the sensor only updates every 30 minutes. You can manually trigger an update by restarting Home Assistant. Keep in mind that 1 request is used for every sensor.

The unit system that the sensor will use is based on the country you set in your Fitbit profile.

> ## Notes and troubleshooting

Reviewer: ```suggestion
## Troubleshooting
```

### Resetting a broken or incorrect configuration

> 

Reviewer: I think the Symptom/Description/Resolution pattern would fit nicely here. 
For an example, see https://github.com/home-assistant/home-assistant.io/blob/current/source/_integrations/_integration_docs_template.markdown?plain=1#L171

> If the Fitbit integration was initially configured incorrectly, simply removing and re-adding the integration may not be sufficient. When removing the integration, Home Assistant asks whether the existing application credentials should be kept or removed.

Reviewer: How do I know I have a broken configuration? What is the symptom? Do I get an error message? If so, I think it should be mentioned here.

If you choose to keep them, the previously stored (and possibly incorrect) credentials will continue to be used automatically during the next setup attempt, which can lead to repeated connection failures.

To fully reset the configuration, the stored application credentials must also be removed (see [Deleting application credentials](/integrations/application_credentials/#deleting-application-credentials)).

### “Connection failed” after authorization

> 

Reviewer: ```suggestion

#### Symptom

```

> In some cases, the authorization appears to succeed, but Home Assistant returns a **Connection failed** error after redirecting back from Fitbit.

Reviewer: ```suggestion
In some cases, the authorization appears to succeed, but Home Assistant returns a Connection failed` error after redirecting back from Fitbit.

The Home Assistant Core log typically shows: the message Failed to fetch user profile for Fitbit API: Error from Fitbit API`.
```
This section sounds like part of the symptom description.

> 

Reviewer: ```suggestion

#### Description

```

> This is usually caused by missing permissions. The Fitbit integration requires the **Profile** scope. If **Profile** is not enabled during authorization, Home Assistant cannot fetch the user profile and the setup fails.

Reviewer: ```suggestion
This error is usually caused by missing permissions. The Fitbit integration requires the **Profile** scope. If **Profile** is not enabled during authorization, Home Assistant cannot fetch the user profile, and the setup fails.
```

Author: Hmm, when I click "Commit changes" I currently get a 500 error from GitHub. I'll try again later. Unfortunately, I'm not 100% familiar with GitHub, so I hope I confirmed everything correctly. It's just important to me that the info appears on the Fitbit integration page, because I had some issues setting up Fitbit correctly without this information.

Author: I can’t apply it because I’m getting the error: “There was an error trying to commit changes: Failed to submit suggested changes.”

The Home Assistant Core log typically shows:

> > Failed to fetch user profile for Fitbit API: Error from fitbit API

Reviewer: ```suggestion
```

> Enable **Profile** in the Fitbit permission settings and repeat the authorization.

Reviewer: ```suggestion
#### Resolution

1. In the Fitbit permission settings, enable **Profile**.
2. Repeat the authorization steps.
```

The unit system that the sensor will use is based on the country you set in your Fitbit profile.

## Troubleshooting

### Resetting a broken or incorrect configuration

If the Fitbit integration was initially configured incorrectly, simply removing and re-adding the integration may not be sufficient. When removing the integration, Home Assistant asks whether the existing application credentials should be kept or removed.

If you choose to keep them, the previously stored (and possibly incorrect) credentials will continue to be used automatically during the next setup attempt, which can lead to repeated connection failures.

To fully reset the configuration, the stored application credentials must also be removed (see [Deleting application credentials](/integrations/application_credentials/#deleting-application-credentials)).

### “Connection failed” after authorization

> #### Symptom

Reviewer: ```suggestion
#### Symptom

```

In some cases, the authorization appears to succeed, but Home Assistant returns a `Connection failed` error after redirecting back from Fitbit.

The Home Assistant Core log typically shows the message `Failed to fetch user profile for Fitbit API: Error from Fitbit API`.
#### Description

> 

Reviewer: ```suggestion

#### Description

```

The unit system that the sensor will use is based on the country you set in your Fitbit profile.

## Troubleshooting

### Resetting a broken or incorrect configuration

If the Fitbit integration was initially configured incorrectly, simply removing and re-adding the integration may not be sufficient. When removing the integration, Home Assistant asks whether the existing application credentials should be kept or removed.

If you choose to keep them, the previously stored (and possibly incorrect) credentials will continue to be used automatically during the next setup attempt, which can lead to repeated connection failures.

To fully reset the configuration, the stored application credentials must also be removed (see [Deleting application credentials](/integrations/application_credentials/#deleting-application-credentials)).

### “Connection failed” after authorization

#### Symptom

> 

Reviewer: ```suggestion
In some cases, the authorization appears to succeed, but Home Assistant returns a `Connection failed` error after redirecting back from Fitbit.

The Home Assistant Core log typically shows the message `Failed to fetch user profile for Fitbit API: Error from Fitbit API`.
```

#### Description

In some cases, the authorization appears to succeed, but Home Assistant returns a `Connection failed` error after redirecting back from Fitbit.

> The Home Assistant Core log typically shows the message `Failed to fetch user profile for Fitbit API: Error from Fitbit API`.

Reviewer: ```suggestion
```
